osquery/core/init.cpp

```diff
@@ -642,13 +642,13 @@ class AlarmRunnable : public InterruptableRunnable {
 
 /// Graceful shutdown request conditional var.
 std::condition_variable kShutdownRequestCV;
+std::mutex kShutdownRequestMutex;
+bool kShutdownRequested{false};
```

Comment:
I am wondering if this ought to be `volatile` bool kShutdownRequested so a dirty value isn't picked up from cache?

Reply:
`volatile` in C++ it's different than in C, it shouldn't be used in the multithreaded case; `std::atomic` it's what one should use.
But in this case it's not required because kShutdownRequested is always accessed (and should always be accessed) behind a mutex.

Reply:
Well, bummer. It looks like `volatile` keyword behaves differently if using ISO-spec or not (https://docs.microsoft.com/en-us/cpp/cpp/volatile-cpp?view=vs-2019). And by default, MSVC will _not_ use the ISO spec but the non-portable version (given it will likely surprise people if they update behavior to ISO spec).

I learned something new, lol.